bundles/org.openhab.binding.heos/src/main/java/org/openhab/binding/heos/internal/handler/HeosBridgeHandler.java

```diff
@@ -0,0 +1,461 @@
+import org.openhab.binding.heos.internal.json.dto.HeosEvent;
+import org.openhab.binding.heos.internal.json.dto.HeosEventObject;
+import org.openhab.binding.heos.internal.json.dto.HeosResponseObject;
+import org.openhab.binding.heos.internal.json.payload.Group;
+import org.openhab.binding.heos.internal.json.payload.Media;
+import org.openhab.binding.heos.internal.json.payload.Player;
+import org.openhab.binding.heos.internal.resources.HeosEventListener;
+import org.openhab.binding.heos.internal.resources.Telnet;
+import org.openhab.binding.heos.internal.resources.Telnet.ReadException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link HeosBridgeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Johannes Einig - Initial contribution
+ */
+@NonNullByDefault
+public class HeosBridgeHandler extends BaseBridgeHandler implements HeosEventListener {
+    private final Logger logger = LoggerFactory.getLogger(HeosBridgeHandler.class);
+
+    private static final int HEOS_PORT = 1255;
+
+    private List<HeosPlayerDiscoveryListener> playerDiscoveryList = new CopyOnWriteArrayList<>();
+    private List<String[]> selectedPlayerList = new CopyOnWriteArrayList<>();
+    private HeosChannelManager channelManager = new HeosChannelManager(this);
+    private HeosChannelHandlerFactory channelHandlerFactory;
+
+    private Map<String, HeosGroupHandler> groupHandlerMap = new ConcurrentHashMap<>();
+    private Map<String, String> hashToGidMap = new ConcurrentHashMap<>();
+
+    private @Nullable ScheduledFuture<?> startupFuture;
+
+    private HeosSystem heos;
+    private @Nullable HeosFacade apiConnection;
+
+    private boolean loggedIn = false;
+    private boolean bridgeHandlerDisposalOngoing = false;
+
+    private @NonNullByDefault({}) BridgeConfiguration configuration;
+
+    private int failureCount;
+
+    public HeosBridgeHandler(Bridge thing, HeosDynamicStateDescriptionProvider heosDynamicStateDescriptionProvider) {
+        super(thing);
+        heos = new HeosSystem(scheduler);
+        channelHandlerFactory = new HeosChannelHandlerFactory(this, heosDynamicStateDescriptionProvider);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            return;
+        }
+        Channel channel = this.getThing().getChannel(channelUID.getId());
+        if (channel == null) {
+            logger.debug("No valid channel found");
+            return;
+        }
+
+        ChannelTypeUID channelTypeUID = channel.getChannelTypeUID();
+        HeosChannelHandler channelHandler = channelHandlerFactory.getChannelHandler(channelUID, this, channelTypeUID);
+        if (channelHandler != null) {
+            try {
+                channelHandler.handleBridgeCommand(command, thing.getUID());
+                failureCount = 0;
+                updateStatus(ONLINE);
```

Comment:
Shouldn't this be
`updateStatus(ThingStatus.ONLINE)`

Reply:
I used a static import here to keep the line a little bit more compact:

https://github.com/openhab/openhab-addons/blob/d56dd54562a2044cd95d09feaee1e2b76b1ad8da/bundles/org.openhab.binding.heos/src/main/java/org/openhab/binding/heos/internal/handler/HeosBridgeHandler.java#L15